feathr_project/feathr/registry/_feathr_registry_client.py

```diff
@@ -0,0 +1,631 @@
+            """
+// THIS FILE IS AUTO GENERATED. PLEASE DO NOT EDIT.
+anchors: {
+    {% for anchor in feature_anchors %}
+        {% if not anchor.source.name == "PASSTHROUGH" %}
+            {{anchor.to_feature_config()}}
+        {% endif %}
+    {% endfor %}
+}
+
+sources: {
+    {% for source in sources%}
+        {% if not source.name == "PASSTHROUGH" %}
+            {{source.to_feature_config()}}
+        {% endif %}
+    {% endfor %}
+}
+"""
+        )
+        anchored_feature_configs = tm.render(feature_anchors=repo_definitions.feature_anchors,
+                                             sources=repo_definitions.sources)
+        config_file_path = os.path.join(local_workspace_dir, config_file_name)
+        write_to_file(content=anchored_feature_configs,
+                      full_file_name=config_file_path)
+
+    @classmethod
+    def _save_derived_feature_config(self, repo_definitions: RepoDefinitions, local_workspace_dir="./"):
+        config_file_name = "feature_conf/auto_generated_derived_features.conf"
+        tm = Template(
+            """
+anchors: {}
+derivations: {
+    {% for derived_feature in derived_features %}
+        {{derived_feature.to_feature_config()}}
+    {% endfor %}
+}
+"""
+        )
+        derived_feature_configs = tm.render(
+            derived_features=repo_definitions.derived_features)
+        config_file_path = os.path.join(local_workspace_dir, config_file_name)
+        write_to_file(content=derived_feature_configs,
+                      full_file_name=config_file_path)
+
+
+def check(r):
+    if not r.ok:
+        raise RuntimeError(
+            f"Failed to call registry API, status is {r.status_code}, error is {r.text}")
+    return r
+
+
+def source_to_def(v: Source) -> dict:
+    ret = {}
+    if v.name == INPUT_CONTEXT:
+        return {
+            "name": INPUT_CONTEXT,
+            "type": INPUT_CONTEXT,
+            "path": INPUT_CONTEXT,
+        }
+    elif isinstance(v, HdfsSource):
+        ret = {
+            "name": v.name,
+            "type": urlparse(v.path).scheme,
+            "path": v.path,
+        }
+    elif isinstance(v, JdbcSource):
+        ret = {
+            "name": v.name,
+            "type": "jdbc",
+            "url": v.url,
+        }
+        if hasattr(v, "dbtable") and v.dbtable:
+            ret["dbtable"] = v.dbtable
+        if hasattr(v, "query") and v.query:
+            ret["query"] = v.query
+        if hasattr(v, "auth") and v.auth:
+            ret["auth"] = v.auth
+    else:
+        raise ValueError(f"Unsupported source type {v.__class__}")
+    if v.preprocessing:
+        ret["preprocessing"] = v.preprocessing
+    if v.event_timestamp_column:
+        ret["eventTimestampColumn"] = v.event_timestamp_column
+    if v.timestamp_format:
+        ret["timestampFormat"] = v.timestamp_format
+    if v.registry_tags:
+        ret["tags"] = v.registry_tags
+    return ret
+
+
+def dict_to_source(v: dict) -> Source:
+    id = UUID(v["guid"])
```

Comment:
Should this be an internal method so should start with _? Also add a bit comments for the purpose of this method.